### DISCUSSION TITLE
The activation status of CodeQL is unclear

### DISCUSSION THREAD

**Discussion by ferferga** [community/community#55038] [Code Security]
### Select Topic Area

Product Feedback

### Body

Hello

I consider myself an advanced GitHub user. I've been using CodeQL since the early days. However, since it's getting more and more integrated into the repository settings, I found it more confusing to use. Here's what I have I'm greeted with in the repository settings:

![image](https://github.com/community/community/assets/10274099/7ad225ee-f18b-4c47-8a6d-63be50263d6f)

As you can see, there's a "Set up" button, **which led me to believe a few times that CodeQL is not enabled in my repository and I won't get any alerts if the workflows detect them**. I renamed the workflow file to exactly match ``codeql.yml`` (since I had another name for it), thinking the issue was the names didn't match, but the "Set up" button is still there. **So, I assume so it will always be there when using custom configurations and not the default one**.

Here's another screenshot at `https://github.com/{{repo-path}}/security/code-scanning/tools/CodeQL/status` that demonstrates how everything is set up properly, so it's not in my end:
![image](https://github.com/community/community/assets/10274099/193e0d7b-48d8-4723-b217-1bc9d7b29ce2)

In case it's relevant, my repository [is public](https://github.com/jellyfin/jellyfin-vue/).

**I believe this button should disappear regardless the default or custom configurations are being used, since it is really confusing, leading users to believe CodeQL is not configured at all when it indeed does!**

**Comment by AlonaHlobina** [comment]
Hi @ferferga ,
Thank you very much for your detailed and beneficial feedback. The code scanning team has made it a priority to work on improving the confusing experience you described in the upcoming quarters. We greatly appreciate your input; it will help us provide a better user experience.

cc @asiermartinez and @cannist for visibility
